conftest.py

```diff
@@ -161,3 +162,16 @@ def enable_mlflow_testing():
     with pytest.MonkeyPatch.context() as mp:
         mp.setenv(_MLFLOW_TESTING.name, "TRUE")
         yield
+
+
+@pytest.fixture(scope="module", autouse=True)
+def disable_get_bucket_location(request):
```

Comment:
Previously I tried an alternative option of just fixing the tests that were broken by trying to fetch the bucket location, which were the tests that didn't use the `mock_s3_bucket` and tried to actually call methods on the `S3ArtifactRepository`. I've pivoted to this approach as it prevents requiring future s3 tests to have to use the properly mocked s3 buckets. The downside though is that many of the tests that actually use the `S3ArtfiactRepository` won't test for getting the bucket location.